### PR TITLE
Feature/zap logging library implementationrary)

### PR DESCRIPTION
**Pull request by panbhatt**
This feature will implement the Logging in SpinUp. zap library has been introduced to do all the work. 
Two Appendars has been added. 
1. console Appender. 
2. File Appender : This would be configured via Spinup.yaml configuration file. Please see a sample file with two new attributes "log_dir" and "log_file".
----------------------------------
common:
  ports: [
    5432, 5433, 5434, 5435, 5436, 5437
  ]
  architecture: amd64
  projectDir: /home/pankaj/GolangProjects/spinup
  client_id: <CLIENT_ID>
  client_secret: <CLIENT_SECRET>
  api_key: abcd
  log_dir: /home/pankaj
  log_file: spin.log
------------------------------------
All the statements like log.FATAL and fmt.Println/Printf needs to be replaed by "utils.Logger.Info("MESSAGE") and errors would need to be replaced by "utils.Logger.Error("something weird occured", zap.Error("err", err))

Please see examples in backup.go  and start.go. 
we need to make sure to use Logger only when the application has been booted correctly, otherwise the startup errors can be lost (if zap not being configured properly). 

This implementation did not changed all the logging statements in all the files, only in some of the files. we will do it one by one and then we need to cut out a release.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -19,6 +19,8 @@ type Configuration struct {
 		ClientID     string `yaml:"client_id"`
 		ClientSecret string `yaml:"client_secret"`
 		ApiKey       string `yaml:"api_key"`
+		LogDir       string `yaml:"log_dir"`
+		LogFile       string `yaml:"log_file"`
```
Comment (viggy28) [inline config/config.go:23]: Nit: an extra space here.

File: internal/backup/backup.go [hunk 2]
```diff
@@ -108,28 +110,29 @@ func CreateBackup(w http.ResponseWriter, r *http.Request) {
...
 	dockerClient, err := dockerservice.NewDocker()
 	if err != nil {
-		fmt.Printf("error creating client %v", err)
+		utils.Logger.Error("Error creating client", zap.Error(err))
+		
 	}
 	pgContainer, err := dockerClient.GetContainer(context.Background(), pgHost)
 	if err != nil {
-		log.Printf("error: getting container for container name %s %v", pgHost, err)
+		utils.Logger.Error("Error getting container for container name ", zap.String("containerName", pgHost), zap.Error(err))
```
Comment (viggy28) [inline internal/backup/backup.go:125]: I prefer not capitalizing errors in general.

File: internal/cmd/start.go [hunk 3]
```diff
@@ -76,15 +79,17 @@ func startCmd() *cobra.Command {
 		Use:   "start",
 		Short: "start the spinup API and frontend servers",
 		Run: func(cmd *cobra.Command, args []string) {
-			log.Println(fmt.Sprintf("INFO: using config file: %s", cfgFile))
+			utils.InitializeLogger("", "")
```
Comment (viggy28) [inline internal/cmd/start.go:82]: do we need this here? Looks like we initializing on line https://github.com/spinup-host/spinup/pull/115/files#diff-032043c7b9694777926f5fa9af41a34f3df0230518bf749b08df4f2a47fecd32R88

**Review by viggy28 — APPROVED** [review]

File: internal/cmd/start.go [hunk 4]
```diff
@@ -93,35 +98,37 @@ func startCmd() *cobra.Command {
...
-					log.Fatalf("FATAL: starting UI server %v", err)
+					utils.Logger.Fatal("starting UI server", zap.Error(err))
 				}
 
 				uiServer := &http.Server{
 					Handler: uiHandler(),
 				}
 				go func() {
-					log.Println(fmt.Sprintf("INFO: starting Spinup UI on port %s", uiPort))
+					utils.Logger.Info("sstarting Spinup UI   ", zap.String("port", uiPort))
```
Comment (viggy28) [inline internal/cmd/start.go:115]: s/sstarting/starting

**Review by viggy28 — APPROVED** [review]

File: config/config.go (hunk 1 above)
Comment (idoqo) [inline config/config.go:23]: I noticed we join LogDir with LogFile at some point. What do you think about having this as a single config? We can remove LogDir and make LogFile the absolute path to the log location. Something like:

`log_file: /home/pankaj/spinup.log`

File: internal/cmd/start.go [hunk 5]
```diff
@@ -93,35 +98,37 @@ func startCmd() *cobra.Command {
...
+			utils.Logger.Info(fmt.Sprint(<-stopCh))
+			utils.Logger.Info("stopping spinup apiServer")
+
 		},
 	}
 
 	home, err := os.UserHomeDir()
 	if err != nil {
-		log.Fatalf("FATAL: obtaining home directory: %v", err)
+		utils.Logger.Fatal("Fobtaining home directory: ", zap.Error(err))
```
Comment (idoqo) [inline internal/cmd/start.go:131]:
```suggestion
		utils.Logger.Fatal("obtaining home directory: ", zap.Error(err))
```

**Review by idoqo — COMMENTED** [review]
looks okay, with some general thoughts too.

**Review by skant7 — APPROVED** [review]

File: config/config.go (hunk 1 above)
Comment (viggy28) [reply config/config.go:23]: good point. Less confusion 👍

**Pull request merged**